virtio-devices/src/vhost_user/vu_common_ctrl.rs

```diff
@@ -147,8 +147,14 @@ pub fn setup_vhost_user(
 
         vu.set_vring_addr(queue_index, &config_data)
             .map_err(Error::VhostUserSetVringAddr)?;
-        vu.set_vring_base(queue_index, 0u16)
-            .map_err(Error::VhostUserSetVringBase)?;
+
+        if let Ok(last_avail_index) = queue.avail_index_from_memory(mem) {
+            vu.set_vring_base(queue_index, last_avail_index)
+                .map_err(Error::VhostUserSetVringBase)?;
+        } else {
+            vu.set_vring_base(queue_index, 0u16)
+                .map_err(Error::VhostUserSetVringBase)?;
+        }
```

Comment:
This looks good, but I'd prefer the following syntax:
```rust
let avail_index = if let Ok(index) = queue.avail_index_from_memory(mem) {
    index
} else {
    0
};
vu.set_vring_base(queue_index, avail_index).map_err(Error::VhostUserSetVringBase)?;
```

Reply:
Is it okay to just lose the error in this case? Sounds bad.

Reply:
Well actually we could always get the avail index from memory. Even for the 0 case.
So I think we can simply go with:
```rust
vu.set_vring_base(queue_index, queue.avail_index_from_memory(mem).map_err(Error::GetAvailableIndex)?).map_err(Error::VhostUserSetVringBase)?;
```

Reply:
> Well actually we could always get the avail index from memory. Even for the 0 case.
> So I think we can simply go with:
> 
> ```rust
> vu.set_vring_base(queue_index, queue.avail_index_from_memory(mem).map_err(Error::GetAvailableIndex)?).map_err(Error::VhostUserSetVringBase)?;
> ```

Changed to this form.